src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -278,21 +278,36 @@ protected override TDeclarationNode AddNamespace<TDeclarationNode>(TDeclarationN
             CodeGenerationOptions options,
             CancellationToken cancellationToken)
         {
-            var memberDeclaration = destinationMember as MemberDeclarationSyntax;
-            if (memberDeclaration == null)
-            {
-                return destinationMember;
-            }
+            SyntaxNode finalMember;
 
-            var currentParameterList = memberDeclaration.GetParameterList();
-            if (currentParameterList == null)
+
+            //Also supports local function statement syntax 
+            switch (destinationMember)
             {
-                return destinationMember;
+                case MemberDeclarationSyntax memberDeclaration:
```

Comment:
OK, after looking more closely at the implementation of `CSharpSyntaxGenerator`, it seems both `GetParameterList` and `WithParameterList` are handling local function syntax just fine (also lambdas), we might want to switch to using SyntaxGenerator here instead of those extension methods for `MemberDeclarationSyntax`